### PR TITLE
LSIF Typed: miscellaneous code cleanups

### PR DESCRIPTION
**Pull request by olafurpg**
Without this line, gopls reports an error about broken imports.

## Test plan

If the CI is green then we should be safe to merge.

### CODE REVIEW

**Comment by olafurpg** [comment]
Running `go mod tidy` removes the tree-sitter dependency for me locally, and I suspect the CI will report the same error

**Comment by olafurpg** [comment]
I double-checked to confirm that gopls no longer reports errors if you open VS Code at the root of the repo

File: internal/codeintel/lsiftyped/tree_sitter.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package lsiftyped
+
+import reproLang "github.com/sourcegraph/sourcegraph/lib/codeintel/reprolang/bindings/golang"
+
+// This function only exists so that `go mod tidy` doesn't remove the go-tree-sitter dependency from
+// go.sum at the root of the directory. Without this file, gopls reports an error about broken imports
+// for files in the file "lib/codeintel/reprolang/src/binding.go".
+//lint:ignore U1000 This function is intentionally unused
+func unusedTreeSitter() reproLang.Dependency {
```
Comment (olafurpg) [inline internal/codeintel/lsiftyped/tree_sitter.go:9]: This is an ugly hack. Any ideas for a cleaner solution?

**Review by mrnugget — APPROVED** [review]
No idea about the hack yet. I'm still unsure why we get the error, since nothing outside `lib` imports it.

**Comment by olafurpg** [comment]
My guess is that our non-lib code transitively depends on everything inside lib/go.sum except the newly introduced reprolang parser.

**Comment by olafurpg** [comment]
I'll go ahead and merge since this fixes a gopls error for everybody who are working with the latest main

File: lib/codeintel/reprolang/bindings/golang/dependency.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package golang
+
+import "github.com/sourcegraph/sourcegraph/lib/codeintel/lsiftyped"
+
+type Dependency struct {
```
Comment (efritz) [inline lib/codeintel/reprolang/bindings/golang/dependency.go:5]: Everything but this looks like a rename...?

File: internal/codeintel/lsiftyped/tree_sitter.go (hunk 1 above)
Comment (efritz) [reply internal/codeintel/lsiftyped/tree_sitter.go:9]: I think [something like this](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:tools.go&patternType=literal) is what you want.

File: lib/codeintel/reprolang/bindings/golang/dependency.go (hunk 2 above)
Comment (olafurpg) [reply lib/codeintel/reprolang/bindings/golang/dependency.go:5]: This is also a rename if you run `git log -M20 -p --stat`. The package rename + file rename exceeded the default threshold for rename detection

**Pull request merged**